consistency/tool/clustering/main.py

```diff
@@ -0,0 +1,85 @@
+import grpc
+from google.cloud.apigeeregistry.v1 import registry_service_pb2_grpc
+from google.cloud.apigeeregistry.v1 import registry_models_pb2 as rm
+from google.cloud.apigeeregistry.v1 import registry_service_pb2 as rs
+from word_extraction import ExtractWords
+from clustering import ClusterWords
+import argparse
+
+
+def main():
+    # Creating registry client
+    channel = grpc.insecure_channel("localhost:8080")
+    stub = registry_service_pb2_grpc.RegistryStub(channel)
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--project_name",
+        type=str,
+        required=True,
+        help="Name of the project to compute clusters for",
+    )
+
+    args = parser.parse_args()
+
+    project_name = args.project_name
+
+    extrct = ExtractWords(stub=stub, project_name=project_name)
+    try:
+        words = extrct.get_vocabs()
+
+    except Exception as e:
+        print(e, " \n Getting words failed")
+
+    clustr = ClusterWords(stub=stub, words=words)
+
+    try:
+        clustr.clean_words()
+        word_groups = clustr.create_word_groups()
+    except Exception as e:
+        print(e, " \n Clustering words failed")
+
+    # upload the wordGroups to the server
+
+    for word_group in word_groups:
+
+        id = "".join(filter(str.isalnum, word_group.id.lower()))
+
+        artifact = rm.Artifact(
+            name="projects/"
+            + project_name
+            + "/locations/global/artifacts/wordgroup-"
+            + id,
+            mime_type="application/octet-stream;type=google.cloud.apigeeregistry.applications.v1alpha1.consistency.WordGroup",
+            contents=word_group.SerializeToString(),
+        )
+
+        createArtifactRequest = rs.CreateArtifactRequest(
+            parent="projects/" + project_name + "/locations/global",
+            artifact=artifact,
+            artifact_id="wordgroup-" + id,
```

Comment:
It maybe unrelated to this CL but as we are generating the artifact ID from the cluster ID how would we handle if the cluster ID representing the cluster changes while recomputing clusters? Would we delete the old word group artifact?

Reply:
My understanding is that at each re-computation of clusters, we delete all wordgroups we have stored first. @shrutiparabgoogle may clarify here.

Reply:
Yes, that is correct cluster re-computation will be delete and create initially, next phase can be to make it smart by deleting only non-existent clusters and updating the existing ones.